v22.2/common-table-expressions.md

```diff
@@ -295,7 +295,155 @@ SELECT * FROM cte LIMIT 10;
 (10 rows)
 ~~~
 
-While this practice works for testing and debugging, Cockroach Labs does not recommend it in production.
+While adding a limit to prevent infinite recursion works for testing and debugging, Cockroach Labs does not recommend it in production. It is best practice to ensure that recursive subqueries have an explicit end condition.
+
+### Loose index scan using a recursive CTE
+
+You can use a recursive CTE to perform a loose index scan, which speeds up certain queries that would otherwise require a full scan. A loose index scan reads noncontiguous ranges of an index by performing multiple shorter scans.
+
+Create a table:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE TABLE test (n INT);
+~~~
+
+Populate the table with many random values from 0 to 9:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+INSERT INTO test SELECT floor(random() * 10)
+FROM generate_series(1, 1000000);
+~~~
+
+Create an index:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE INDEX ON test (n);
+~~~
+
+Issue a statement to count the number of distinct values:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT COUNT(DISTINCT n) FROM test;
+~~~
+
+~~~
+SELECT COUNT(DISTINCT n) FROM test;
+  count
+---------
+     10
+(1 row)
+
+
+Time: 273ms total (execution 273ms / network 0ms)
+~~~
+
+This statement has a high latency because it reads every row in the index.
+
+{% include_cached copy-clipboard.html %}
```

Comment:
The two `EXPLAIN` statements aren't necessarily introduced. I like the sentences that precede them as explanation for the statement, but an additional "the following `EXPLAIN` statement analyzes x...:" This might help lead the user through a little more. This could be a me thing, so totally optional.